lib/jormungandr/src/Cardano/Wallet/Jormungandr/Network.hs

```diff
@@ -163,11 +181,60 @@ mkNetworkLayer j = NetworkLayer
 
     }
   where
+    getTipId' = (getTipId j) `mappingError`
+        ErrNetworkTipNetworkUnreachable
+    getBlock' t = (getBlock j t) `mappingError` \case
+        ErrGetBlockNotFound _ ->
+            ErrNetworkTipNotFound
+        ErrGetBlockNetworkUnreachable e ->
+            ErrNetworkTipNetworkUnreachable e
+
     mappingError = flip withExceptT
 
     tailOrEmpty [] = []
     tailOrEmpty (_:xs) = xs
 
+{-------------------------------------------------------------------------------
+                                 State munching
+-------------------------------------------------------------------------------}
+
+-- | A list of blocks and their hashes.
+-- The first block in this sequence is the network tip.
+-- The last block in this sequence is the block of depth _k_,
+-- which is the last unstable block.
+newtype UnstableBlocks = UnstableBlocks
+    { getUnstableBlocks :: Seq (Hash "BlockHeader", Block Tx) }
+    deriving (Show, Eq, Semigroup, Monoid)
```

Comment:
small detail but, when using `GeneralizedNewtypeDeriving`, let's also always use `DerivingStrategies` to make things clear in case we suddenly bring `DeriveAnyClass` or `DerivingVia` into play. It makes things a bit more readable.

Reply:
OK fixed.